File: 0118-pascals-triangle/Solution.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> triangle = new ArrayList<List<Integer>>();
        List<Integer> lastRow = new ArrayList<>();

        for (int x = 0; x < numRows; x++) {
            List<Integer> currRow = new ArrayList<>();

            for (int z = 0; z <= x; z++) {
                if (z == 0 || z == x) {
                    currRow.add(1);
                    continue;
                }

                currRow.add(lastRow.get(z) + lastRow.get(z - 1));
            }

            lastRow = currRow;
            triangle.add(currRow);
        }

        return triangle;
    }
}

File: 0118-pascals-triangle/Solution.js
var generate = function (numRows) {
    let triangle = [];
    let lastRow = [];

    for (let i = 0; i < numRows; i++) {
        let currRow = [];

        for (let j = 0; j <= i; j++) {
            if (j == 0 || j == i) {
                currRow.push(1);
                continue;
            }

            currRow.push(lastRow[j] + lastRow[j - 1]);
        }

        lastRow = currRow.slice();
        triangle.push(currRow);
    }

    return triangle;
};
